packages/recoil/recoil_values/Recoil_atom.js

```diff
@@ -292,6 +293,23 @@ function baseAtom<T>(options: BaseAtomOptions<T>): RecoilState<T> {
         return getLoadable(recoilValue).toPromise();
       }
 
+      function observe<S>(
+        recoilValue: RecoilValue<S>,
+        observer: (value: S) => void,
+      ): () => void {
+        const observed = recoilValue;
+        const {release} = store.subscribeToTransactions(store => {
+          const loadable = store
+            .getState()
+            .currentTree?.atomValues.get(recoilValue.key);
+          if (loadable) {
+            const value = loadable.getValue();
```

Comment:
`getValue()` could throw in error or pending states.  Check that the state is `hasValue` to avoid uncaught exceptions.

Reply:
Yep, makes sense